test/integration/buildrun_cleanup_test.go

```diff
@@ -0,0 +1,454 @@
+
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+			buildObject, err = tb.GetBuildTillValidation(buildObject.Name)
+			Expect(err).To(BeNil())
+
+			// Create first buildrun
+			buildRunObject.Name = BUILDRUN + tb.Namespace + "-1"
+			Expect(tb.CreateBR(buildRunObject)).To(BeNil())
+			br1, err := tb.GetBRTillCompletion(buildRunObject.Name)
+			Expect(err).To(BeNil())
+			Expect(br1.Status.GetCondition(v1alpha1.Succeeded).Status).To(Equal(corev1.ConditionTrue))
+			// Create second buildrun
+			buildRunObject.Name = BUILDRUN + tb.Namespace + "-2"
+			Expect(tb.CreateBR(buildRunObject)).To(BeNil())
+			br2, err := tb.GetBRTillCompletion(buildRunObject.Name)
+			Expect(err).To(BeNil())
+			Expect(br2.Status.GetCondition(v1alpha1.Succeeded).Status).To(Equal(corev1.ConditionTrue))
+
+			_, err = tb.GetBRTillNotFound(BUILDRUN+tb.Namespace+"-1", time.Second*1, time.Second*5)
+			Expect(apierrors.IsNotFound(err)).To(BeTrue())
+			_, err = tb.GetBRTillCompletion(BUILDRUN + tb.Namespace + "-2")
+			Expect(err).To(BeNil())
+		})
+
+	})
+
+	Context("When a buildrun that has TTL defined in its spec and in the corresponding build's spec succeeds", func() {
+
+		BeforeEach(func() {
+			buildSample = []byte(test.MinimalBuildWithRetentionTTLOneMin)
+			buildRunSample = []byte(test.MinimalBuildRunRetentionTTLFive)
+		})
+
+		It("Should honor the TTL defined in the buildrun.", func() {
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+			buildObject, err = tb.GetBuildTillValidation(buildObject.Name)
+			Expect(err).To(BeNil())
+			Expect(tb.CreateBR(buildRunObject)).To(BeNil())
+
+			br, err := tb.GetBRTillCompletion(buildRunObject.Name)
+			Expect(err).To(BeNil())
+			Expect(br.Status.GetCondition(v1alpha1.Succeeded).Status).To(Equal(corev1.ConditionTrue))
+			_, err = tb.GetBRTillNotFound(buildRunObject.Name, time.Second*1, time.Second*15)
+			Expect(apierrors.IsNotFound(err)).To(BeTrue())
+
+		})
+	})
+
+	Context("Multiple buildruns with different build limits for failure and success", func() {
+
+		BeforeEach(func() {
+			Expect(err).To(BeNil())
+			buildSample = []byte(test.MinimalBuildWithRetentionLimitDiff)
+			buildRunSample = []byte(test.MinimalBuildahBuildRunWithExitCode)
+		})
+		// Case with failedLimit 1 and succeededLimit 2.
+		// It ensures that only relevant buildruns are affected by retention parameters
+		It("Should not find the old failed buildrun if the limit has been triggered", func() {
+
+			// Create build
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+			buildObject, err = tb.GetBuildTillValidation(buildObject.Name)
+			Expect(err).To(BeNil())
+
+			// Create 2 successful buildruns
+			buildRunObject.Name = BUILDRUN + tb.Namespace + "-success-1"
+			err = tb.CreateBR(buildRunObject)
+
+			buildRunObject.Name = BUILDRUN + tb.Namespace + "-success-2"
+			err = tb.CreateBR(buildRunObject)
+
+			// Create 1 failed buildrun
+			buildRunObject.Name = BUILDRUN + tb.Namespace + "-fail-1"
+			var str string = "false"
```

Comment:
In case you want it shorter.

```suggestion
			str := "false"
```